### ISSUE TITLE
refactor both gem and crystal code

### ISSUE THREAD

**Issue by MadBomber** [MadBomber/justprep#4]
There is some refactoring that can be of benefit.  For example, create a method to standardize the error message format.

Also it might be interesting to extract out those methods between ruby and crystal that have common source code to both.

**Comment by MadBomber** [comment]
I did a quick test to see if a common directory of files could be used between the two build processes.  Short answer is yes.

At the root directory of repo I created a directory "common" into which I placed a file named "version.crb" where the extension ".crb" is my convention for a file that can be used my both Crystal and Ruby.

In the Ruby file, within the module definition I added:
  load COMMON_DIR + "version.crb"

In the crystal directory in its justfile I added "../common/*.crb" to the crystal compile command.

running ruby/bin/justprep --version
produced the same result as
running crystal/bin/justprep --version

This means that I can start pulling out code that is common to both implementations into the common directory.  Code that is specific to an implementation can go into that implementation's lib directory with a normal extension for that implementation.

**Comment by MadBomber** [comment]
I chose to do this issue in the same branch "add_blog" but I ran into a problem with the ruby gem installation process.  With the "common" directory at the repository root, when the ruby gem is installed no file outside the gem's root directory is installed making it unusable.

The Crystal version does not really care where the common directory is located within the repository.  So I might have to mve the common directory from the repository root to ./ruby/lib/common  and rewrite the load commands in the justprep.rb file.  I will also have to rewrite the crystal build recipe in the top level justfile.

Also the way I was using the "bump" gem to manage the VERSION value in two different files does not work when you change the filename to something not ruby.  For example I am use the file extension ".crb" for files in the "common" directory for files that work as is in both ruby and crystal.  I have one file "common/constants.crb" where the VERSION constant is defined.  Bump would not work with that file name.  I had to rewrite the version recipies to use some bash code with grep and sed to rewrite that VERSION value in that specific file.

I'm going to think a little longer on how to reorganize the location of the common directory such that the gem install process for ruby picks up all of the common files.  Crystal does not care where the directory is located since that source is turned into binary during the build process and stuck into the crystal/bin/justprep file.
